CASSANDRA-17317: January 2022 blog "Tightening Security for Apache Cassandra Part: 1" (#94)
apache/cassandra-website: site-content/source/modules/ROOT/pages/blog/Tightening-Security-for-Apache-Cassandra-Part-1.adoc

= Tightening Security for Apache Cassandra: Part 1
:page-layout: single-post
:page-role: blog-post
:page-post-date: January, 31 2022
:page-post-author: Maulin Vasavada
:description: The Apache Cassandra Community

image::blog/tighten-security-p1-unsplash.jpg[secure lock and chain across a door]

Image credit: https://unsplash.com/@thommilkovic[Thom Milkovic on Unsplash^]

This series will show you how Apache Cassandra 4.0+ enables users to customize SSL/TLS configuration flexibly and enhance the database’s security posture. First, we will start with some context before diving into the technical details.

> === According to eMarketer’s

Reviewer: The start of the paragraph as the header doesn't look right on the page so I suggest using a different heading:

```suggestion
=== e-Commerce to grow double digits
```

> https://www.emarketer.com/content/us-ecommerce-forecast-2021[forecast in 2021^]: “US e-commerce sales are projected to continue to grow by double digits, up 17.9% in 2021 to $933.30 billion. E-commerce penetration will continue to increase, more than doubling from 2019 to 23.6% in 2025.” While Emarketer’s data is only for the US, the global trend is seeing a similar upward swing. With this growth in buying online, there is an increased focus on security and how to protect customers’ data and secure communication between computers, especially with databases such as Apache Cassandra that store customer and transaction data.

Reviewer: ```suggestion
According to eMarketer’s https://www.emarketer.com/content/us-ecommerce-forecast-2021[forecast in 2021^]: “US e-commerce sales are projected to continue to grow by double digits, up 17.9% in 2021 to $933.30 billion. E-commerce penetration will continue to increase, more than doubling from 2019 to 23.6% in 2025.” While eMarketer’s data is only for the US, the global trend is seeing a similar upward swing. With this growth in buying online, there is an increased focus on security and how to protect customers’ data and secure communication between computers, especially with databases such as Apache Cassandra that store customer and transaction data.
```

image::blog/emarketer-chart.png[eMarketer chart]
[#img-ecommerce]

> The growth in ecommerce also demands software thats both secure and scalable.

Reviewer: Missing apostrophe:

```suggestion
The growth in ecommerce also demands software that's both secure and scalable.
```

=== Adoption of Apache Cassandra & Securing Data

> Apache Cassandra is the open source NoSQL database for mission-critical data. Its adoption grows day-by-day in the industry, and it’s used by all sizes of organizations serving varied technical and business domains, such as IT Financial Services, Healthcare, Retail, Government, and Education, to name a few. You can find some https://cassandra.apache.org/_/blog/Apache-Cassandra-Usage-Report-2020.html[interesting statistics^] from the 2020 survey about Apache Cassandra’s usage in the industry.

Reviewer: ```suggestion
Apache Cassandra is the open source NoSQL database for mission-critical data. Its adoption grows day-by-day in the industry, and it’s used by all sizes of organizations serving varied technical and business domains, such as IT Financial Services, Healthcare, Retail, Government, and Education, to name a few. You can find some xref:blog/Apache-Cassandra-Usage-Report-2020.adoc[interesting statistics] from the 2020 survey about Apache Cassandra’s usage in the industry.
```

As the web traffic grows for business-to-consumer interactions, service providers need to ensure customer data is securely protected and the backing software can scale to handle growing demand without outages. Service providers use various technologies to support their business functions such as Online Transaction Processing (OLTP), serving internal analytical needs for customer insights, running risk and fraud detection systems, etc. As traffic grows, each sub-system and technology will need to scale with traffic while satisfying security requirements.

Primarily there are two avenues to consider for securing data: data in transit and data at rest.

Data in transit refers to the data traveling between computers over the network (sometimes referred as ‘over-the-wire’). For example, a customer’s credit card details travel to the service provider for payment processing, and, of course, customers expect that to be done securely.

> Data at rest refers to the data digitally stored for durability. Like the Data in transit, the Data at rest needs to be secured with appropriate access control mechanisms and data encryption.

Reviewer: ```suggestion
Data at rest refers to the data digitally stored for durability. Like data in transit, data at rest needs to be secured with appropriate access control mechanisms and data encryption.
```

> In this series, we’re focused on securing data in transit. We will cover the security aspects in more detail in Part 2, but first, a few basics of security followed by how Apache Cassandra enables operators with configurable options for securing data-in-transit, meaning between client nodes and Cassandra server nodes and from server to server.

Reviewer: For consistency, we shouldn't use the hyphenated version of the phrase:

```suggestion
In this series, we’re focused on securing data in transit. We will cover the security aspects in more detail in Part 2, but first, a few basics of security followed by how Apache Cassandra enables operators with configurable options for securing data in transit, meaning between client nodes and Cassandra server nodes and from server to server.
```
